dev-guide/test_framework.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: index
+title: GCAM Testing Framework
+gcam-version: v5.2
+devguide: dev-guide.html
+---
+## GCAM Testing Framework
+This page is here to serve as guidance for GCAM developers who may need to update or extend the GCAM Testing Framework.  Note the GCAM testing framework is meant to automate many of the steps developers must do to verify their changes do not break GCAM on any of the platforms GCAM is known to run on.  The services we use to do this are provided internal to PNNL, as noted below, therefore can not be used outside of the network.
+
+### Notes on initializing and updating the testing framework
+Most users will interact with the testing framework only through the pull request interface and will not need to worry about updating the testing framework.  Some instances of when they would include:
```

Comment:
maybe instead of "when they would" say "when they would need to update" for clarity